nautobot/docs/development/apps/api/platform-features/populating-extensibility-features.md

```diff
@@ -59,3 +63,61 @@ config = AnimalSoundsConfig
     It is crucial that you add the `sender=self` parameter to the `connect` method call - otherwise your signal handler will run as many times as the `NautobotAppConfig.ready` method is called, which may be a lot of times.
 
 After writing this code, run `nautobot-server migrate` or `nautobot-server post_upgrade`, then restart the Nautobot server, and you should see that this custom Relationship has now been automatically created.
+
+## Advanced
+
+While the signal handler approach works for most use cases, it does have its downsides. Other options for the population of base state are available, each with their own trade-offs. There is no single best option, you have to choose the one has the best properties for your specific need.
+
+- Signal Handlers on `nautobot_database_ready` (see above)
+- [Django Onboard Tooling](https://docs.djangoproject.com/en/4.2/howto/initial-data/)
+    - [Django Data Migrations](https://docs.djangoproject.com/en/4.2/topics/migrations/#data-migrations)
+    - [Django Fixtures](https://docs.djangoproject.com/en/4.2/topics/db/fixtures/#fixtures)
+- [Design Builder Jobs](https://docs.nautobot.com/projects/design-builder/en/latest/)
+- Creating data in-place where needed
+
+The following table details points that are somewhat easily comparable between the different options.
+
+| Option                                       | Startup Impact                       | Execution | Performance | Re-running | Idempotence | Modifications Possible      | Custom App Needed |
+|----------------------------------------------|--------------------------------------|-----------|-------------|------------|-------------|-----------------------------|-------------------|
+| Signal Handlers on `nautobot_database_ready` | Exceptions prevent container startup | Automatic | Average     | Possible   | Enforced    | Overwrite/crash on next run | Yes               |
```

Comment:
Seems like the `Re-running` and `Modifications Possible` columns always align - should they be combined into one column?

Reply:
Done!